scipy/sparse/_coo.py

```diff
@@ -652,7 +653,21 @@ def _matmul_vector(self, other):
 
 
     def _matmul_dispatch(self, other):
-        if self.ndim < 3:
+        if not (issparse(other) or isdense(other) or isscalarlike(other)):
```

Comment:
Can we quickly handle scalars up front to simplify this logic? Something like:
```python
        if isscalarlike(other):
            return self * other
```

Reply:
Done at f01ba88873d4c333319e28c21d9f6f6e7cb62d16.